lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -110,6 +110,26 @@ def reload( self, trans, id, **kwd ):
 
     @expose_api
     @web.require_admin
+    def list_requirements(self, trans, **kwds):
+        """
+        GET /api/tools/list_requirements
+        Return list of unique requirements for all tools.
+        """
+
+        return trans.app.toolbox.tools_requirements()
+
+    @expose_api
+    @web.require_admin
+    def tool_requirements_status(self, trans, id, **kwds):
+        """
+        GET /api/tools/{tool_id}/requirement_status
```

Comment:
Let me go through this, but my idea here was to return something like
```python
 [{name='r-deseq2', 'resolver': 'CondaDependencyResolver'}, 'exact': True, {name='r-edger', 'resolver': 'CondaDependencyResolver',  exact=False} ]
```
I could rename this to `requirements_status`, but I like the status in the name ...